### ISSUE TITLE
Documentation of RDF Cube Shape Requirements

### ISSUE THREAD

**Issue by herrstucki** [visualize-admin/visualization-tool#126]
To make it easier for others to implement pipelines that create "valid" (i.e. valid for the visualization tool) cubes we should document the shape properties that are expected.

**Comment by herrstucki** [comment]
These are the main ones currently in use for **dimensions**:

- [dataKind](https://github.com/zazuko/rdf-cube-schema-viz#datakind-temporal--spatial): **Required for temporal dimensions** (spatial not implemented yet)
  - timeUnits supported: unitYear, unitMonth, unitWeek, unitDay, unitHour, unitMinute, unitSecond
- [dataType](https://github.com/zazuko/rdf-cube-schema-viz#datatype-string-boolean-int-float-): **Required** for dimensions containing literal values. Undefined values must be `cube.link/Undefined`.
  - datatypes supported: string, gYear, date, dateTime, in, integer, decimal, float, double
- [scaleType](https://github.com/zazuko/rdf-cube-schema-viz#scaletype-nominal-ordinal-interval-ratio): Only used to distinguish Nominal from Ordinal dimensions 
- [Dimension Types](https://github.com/zazuko/rdf-cube-schema#types-of-dimensions): **Required** to identify Key and Measure Dimensions (SharedDimension not implemented yet)
- `sh:in` (for nominal/ordinal dimensions) or `sh:minInclusive` _and_ `sh:maxInclusive`: **Required**
- `schema:name` **Required**

**Comment by ktk** [comment]
I agree on documentation but we need validation as well so we definitely have to create a SHACL shape that checks for those things. I did that for the cube itself so we should build on that work, see [here](https://github.com/zazuko/rdf-cube-schema/tree/master/validation)

**Comment by Rdataflow** [comment]
@ptbrowne IIUC this documentation has been solved by /docs/ now.
@ktk a valid point, maybe it would help to have cube-creator also validating along these SHACL shapes.

**Comment by ptbrowne** [comment]
Thanks @Rdataflow , yes the shapes and their understanding by visualize is described in detail at : https://visualize.admin.ch/docs/charts/rdf-to-visualize
